packages/opentelemetry-instrumentation-openai/tests/traces/test_responses.py

```
@@ -507,6 +507,234 @@ def test_response_stream_init_with_none_tools():
+            if chunk.type == "response.output_text.delta":
+                full_text += chunk.delta
+
+    # Verify span hierarchy
+    spans = span_exporter.get_finished_spans()
+    parent_spans = [s for s in spans if s.name == "guardrails-wrapper"]
+    openai_spans = [s for s in spans if s.name == "openai.response"]
+
+    assert len(parent_spans) == 1, "Should have exactly one parent span"
+    assert len(openai_spans) == 1, "Should have exactly one OpenAI response span"
+
+    parent = parent_spans[0]
```

Flake8 (7.3.0)

[error] local variable 'parent' is assigned to but never used

(F841)